v1/src/main/java/com/google/cloud/teleport/spanner/ddl/GraphElementTable.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.spanner.ddl;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.spanner.Dialect;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.LinkedHashMap;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class GraphElementTable implements Serializable {
```

Comment:
Can you create and add unit tests for GraphElementTable and PropertyGraph files?

Reply:
Good point. Done. Parts of the new tests overlap with the test in DdlTest, but it also tests each API individually for more coverage.